app/src/main/java/com/github/se/orator/model/speechBattle/BattleRepositoryFirestore.kt

```diff
@@ -321,4 +294,37 @@ class BattleRepositoryFirestore(private val db: FirebaseFirestore) : BattleRepos
           callback(false)
         }
   }
+
+  /** Converts EvaluationResult to a Firestore-friendly map */
+  private fun evaluationResultToMap(evaluation: EvaluationResult): Map<String, Any> {
+    return mapOf(
+        "winnerUid" to evaluation.winnerUid,
+        "winnerMessage" to evaluation.winnerMessage.toMap(),
+        "loserMessage" to evaluation.loserMessage.toMap())
+  }
+
+  /**
+   * Updates the evaluation result for a given battle. This sets the status to COMPLETED, sets the
```

Comment:
Implemented cleanly but along the previous function, it can benefit of a better kdoc documentation. Explain a bit more what you did and add the params in the comment.

Reply:
Thanks for the suggestion